picard/file.py

```diff
@@ -156,9 +156,22 @@ def __init__(self, filename):
     def __repr__(self):
         return '<%s %r>' % (type(self).__name__, self.base_filename)
 
-    @property
-    def new_metadata(self):
-        return self.metadata
+    # pylint: disable=no-self-use
+    def format_specific_metadata(self, metadata, tag, settings=None):
+        """Can be overridden to customize how a tag is displayed in the UI.
+        This is useful if a tag saved to the underlying format will differ from
+        the internal representation in a way that would cause data loss. This is e.g.
+        the case for some ID3v2.3 tags.
+
+        Args:
+            metadata: The metadata object to read the tag from
+            tag: Name of the tag
+            settings: Dictionary of settings. If not set, config.setting should be used
```

Comment:
It's a bit weird that config.setting is recommended here, but the only overridden implementation uses cached_settings instead.

Reply:
Which is just a helper function to read certain keys from config.setting into a dictionary. There is no need to use that, it was just wanted for avoiding code duplication for simar use cases.

The point here is that implementations shod be able to rely on the fact that the passed dict provides all the keys of config.setting, and if none is passed they should hence read the keys from config.setting, which the current one mentation does

Reply:
We don't necessarily need this helper, though. After the discussion basically went full circle I would also be fine with doing this in the ID3 implementation of format_specific_metadata again:

```python
def format_specific_metadata(self, metadata, tag, settings=None):
    if not settings:
        settings = get_config().setting
```

Reply:
I'm ok with either of the following options:

1. Use `settings = get_config().setting` in the ID3 implementation. This is the "I read the docstring of File.format_specific_metadata and implemented the fallback verbatim" version :-)
2. Add a small note to the ID3 implementation saying that `cached_settings` is faster for that case than accessing config.setting - if it still is, I kind of lost track of the various pull requests around config value reading :-) But if it's faster for the ID3 implementation that really reads only one - and in a subset of cases, two - values from config.setting, wouldn't that generally be also true for any other implementation that needs to read at least one value?

Reply:
>     2\. s faster for that case than accessing config.setting - if it still is

It's not faster in that case. Both cases make two reads to config.setting. If several calls to this function are made a dict of settings certainly should be passed. That's also the only use case right now. For single call (should there be one) it is easier to just not pass settings object, in which case there should not really be a performance difference either way.

As this helper function is kind of pointless and maybe ads more confision now I removed it again in add39216bfa3ba58e80b2c9a21205d45e5281212 . @zas Fine with that?

Reply:
Yes, it isn't needed anymore.